### PR TITLE
Implemented OSCForwardTask and made updations to NFBServer

### LINKED ISSUE #209: Missing OscForwardTask
**Actual Behaviour.**  
Missing OscForwardTask

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #209 

**Changes**: Implemented OSCForwardTask and made updations to NFBServer.

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

### CODE REVIEW

**Review by mariobehling — APPROVED** [review]

**Comment by jddeep** [comment]
@mariobehling Please don't merge this right now. Let me implement the OSCForwardMask before this.

**Review by suresh-gandhi — APPROVED** [review]

File: app/src/main/java/io/neurolab/main/NFBServer.java [hunk 1]
```diff
@@ -26,11 +32,30 @@
     private LinkedList<Task> tasks;
     private long lastTimestamp;
 
+    public Config getConfig() {
+        return config;
+    }
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/NFBServer.java]: @jddeep Everything looks good. Just rearrange the code. Getters and setters should be below.

File: app/src/main/java/io/neurolab/main/NFBServer.java [hunk 2]
```diff
@@ -46,6 +71,30 @@ public int getNumSamples() {
...
+    public void setTasks(LinkedList<Task> tasks) {
+        this.tasks = tasks;
+
+        for (Task task : tasks) {
+            if (task instanceof FFTPreprocessor)
+                this.fftData = ((FFTPreprocessor) task).getFFTData();
+            if (task instanceof FileOutputTask)
+                this.fileOutputTask = (FileOutputTask) task;
+        }
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/NFBServer.java]: @jddeep remove the extra space of line.

File: app/src/main/java/io/neurolab/main/task/OscForwardTask.java [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+        rangeValues[index] = maxValues[index] - minValues[index];
+    }
+
+    public synchronized void setOutputString(int index, String string) {
+        outputs[index] = string;
+    }
+
+    // 0 = per channel per bin
+    // 1 = per channel per bin
+    private int forwardMode = 0;
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/task/OscForwardTask.java]: @jddeep rearrange the code.

File: app/src/main/java/io/neurolab/main/task/OscForwardTask.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+        System.err.println("init osc forward");
+    }
+
+    public boolean disconnect() {
+        return oscForwarder.disconnect();
+    }
+
+    @Override
+    public void run() {
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/task/OscForwardTask.java]: remove an extra line of space from here as well.

File: app/src/main/java/io/neurolab/main/task/OscForwardTask.java [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+                    minValues[cb] = minVal;
+                    maxValues[cb] = maxVal;
+                    rangeValues[cb] = maxVal - minVal;
+//                  outputs[cb] = oscForwardMask.getOuputs()[cb].getText();
+                }
+            }
+        }
+
+        if (connected) {
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/task/OscForwardTask.java]: here also

File: app/src/main/java/io/neurolab/main/task/OscForwardTask.java [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+                        msg.addArgument(argument);
+                        bundle.addPacket(msg);
+                    }
+                }
+                oscForwarder.forwardBundle(bundle);
+            } catch (java.lang.IllegalArgumentException e) {
+                e.printStackTrace();
+            }
+        }
+
```
Comment (suresh-gandhi) [inline app/src/main/java/io/neurolab/main/task/OscForwardTask.java]: .

**Review by suresh-gandhi — CHANGES_REQUESTED** [review]

**Review by suresh-gandhi — APPROVED** [review]

**Pull request merged**